internal/provider/webhook_recipient_resource.go

```diff
@@ -332,19 +383,37 @@ func webhookTemplatesToClientPayloads(ctx context.Context, set types.Set, diags
 		}
 	}
 
+	clientVars := make([]client.TemplateVariable, len(variables))
+	for i, v := range variables {
+		tmplVar := client.TemplateVariable{
+			Name:    v.Name.ValueString(),
+			Default: v.DefaultValue.ValueString(),
+		}
+
+		clientVars[i] = tmplVar
+	}
+	clientWebhookPayloads.TemplateVariables = clientVars
+
 	return clientWebhookPayloads
 }
 
-func clientPayloadsToWebhookTemplates(ctx context.Context, p *client.WebhookPayloads, diags *diag.Diagnostics) types.Set {
+func clientPayloadsToSets(ctx context.Context, p *client.WebhookPayloads, diags *diag.Diagnostics) (types.Set, types.Set) {
```

Comment:
nit-ish: these functions are scoped to the whole package so a little more descriptive name would be good, I think
```suggestion
func clientPayloadsToWebhookTemplateSets(ctx context.Context, p *client.WebhookPayloads, diags *diag.Diagnostics) (types.Set, types.Set) {
```